includes/blocks/class-sensei-blocks-initializer.php

```diff
@@ -26,56 +26,50 @@ abstract class Sensei_Blocks_Initializer {
 	 *
 	 * @param array|null $post_types The post types to initialize blocks for, or null for any post type.
 	 */
-	public function __construct( $post_types ) {
-		add_action( 'template_redirect', [ $this, 'maybe_initialize_blocks' ] );
-		add_action( 'current_screen', [ $this, 'maybe_initialize_blocks' ] );
+	public function __construct( $post_types = null ) {
+		add_action( 'init', [ $this, 'maybe_initialize_blocks' ], 200 ); // Should be after all post types are registered.
 
 		$this->post_types = $post_types;
 	}
 
-	/**
-	 * Check if post type is included for initialization.
-	 *
-	 * @param string $post_type Post type to check.
-	 *
-	 * @return boolean Whether post type is included.
-	 */
-	private function is_post_type_included( $post_type ) {
-		return null === $this->post_types || in_array( $post_type, $this->post_types, true );
-	}
-
 	/**
 	 * Check if blocks should be initialized and do initialization.
 	 *
 	 * @access private
 	 */
 	public function maybe_initialize_blocks() {
-		if ( is_admin() ) {
-			$screen = get_current_screen();
-
-			if ( empty( $screen ) ) {
-				return;
-			}
+		if ( ! $this->should_initialize_blocks() ) {
+			return;
+		}
 
-			$is_editor = $screen->is_block_editor || in_array( $screen->id, [ 'widgets', 'site-editor', 'appearance_page_gutenberg-edit-site' ], true );
+		$this->initialize_blocks();
+		$this->initialize_assets();
+	}
 
-			// Init blocks.
-			if ( ! $is_editor || ! $this->is_post_type_included( $screen->post_type ) ) {
-				return;
-			}
-		} elseif ( ! $this->is_post_type_included( get_post_type() ) ) {
-			return;
+	/**
+	 * Initializes the block assets.
+	 */
+	private function initialize_assets(): void {
+		if ( is_admin() ) {
+			add_action( 'enqueue_block_assets', [ $this, 'enqueue_block_assets' ] );
+			add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_assets' ] );
+		} else {
+			add_action( 'template_redirect', [ $this, 'initialize_frontend_assets' ], 9 );
 		}
+	}
 
-		if ( ! $this->should_initialize_blocks() ) {
+	/**
+	 * Initializes the block assets for the frontend.
+	 *
+	 * @access private
+	 */
+	public function initialize_frontend_assets(): void {
```

Comment:
We enqueue editor assets at the end of this method:
```
add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_block_editor_assets' ] );
```
Do we need it here? I see we also enqueue them in the `initialize_assets` for admin pages.

Reply:
Hm, good question. I'm following the previous logic, so I'm guessing it was wrong.

I will remove it from the front-end and see what happens when we get back from the team meetup.

Reply:
The `enqueue_block_editor_assets` hook is not triggering in the front-end so I've removed that line in e9ea21cfdc5fc6fc5516d6a8502777dda16cbf4c. Everything looks to be working as before.